library/std/src/sys/unix/process/process_common.rs

```diff
@@ -202,6 +239,48 @@ impl Command {
         self.gid
     }
 
+    pub fn get_size(&mut self) -> io::Result<usize> {
+        // Envp size calculation is approximate.
+        let env = &self.env;
+        let problem = &mut self.problem;
+        let env_size = self.env_size.get_or_insert_with(|| {
+            let env_map = env.capture();
+            env_map
+                .iter()
+                .map(|(k, v)| {
+                    os2c(k.as_ref(), problem).to_bytes().len()
+                        + os2c(v.as_ref(), problem).to_bytes().len()
+                        + 2
+                })
+                .sum::<usize>()
+                + (env_map.len() + 1) * mem::size_of::<*const u8>()
+        });
+
+        Ok(self.arg_size + *env_size)
+    }
+
+    pub fn check_size(&mut self, refresh: bool) -> io::Result<bool> {
+        use crate::sys;
+        use core::convert::TryInto;
+        if refresh || self.arg_max.is_none() {
+            let (limit, errno) = unsafe {
+                let old_errno = sys::os::errno();
+                sys::os::set_errno(0);
+                let limit = libc::sysconf(libc::_SC_ARG_MAX);
+                let errno = sys::os::errno();
+                sys::os::set_errno(old_errno);
+                (limit, errno)
+            };
+
+            if errno != 0 {
+                return Err(io::Error::from_raw_os_error(errno));
+            } else {
+                self.arg_max = limit.try_into().ok();
+            }
+        }
+        Ok(self.arg_max.unwrap() < 0 || self.get_size()? < (self.arg_max.unwrap() as usize))
```

Comment:
Shouldn't `arg_max` be `usize` so it won't be less than zero?

Reply:
No, a negative value indicates that the arg size is unlimited. If my reading of POSIX is right, that is.

Reply:
IIRC `-0` of `usize` is still `-1` for unlimited.